### ISSUE TITLE
Feature request: parse environment in sgconfig yml files

### ISSUE THREAD

**Issue by hackery** [floragunncom/search-guard#619]
I'm finding use cases here for substituting a few values into a "template" _sg_config.yml_ before handing it to `sgadmin`. The only alternatives seem to be pre-processing, or maintaining multiple variants in SCM, neither of which fit well.

As yaml provides no external substitution mechanism (and there's no string interpolation for aliases) would it be reasonable to add an `sgadmin` option to expand environment variables in the _sgconfig_ files before parsing and sending to the cluster? Linux has the handy `envsubst` utility for pre-processing a file, but I have to hand this to a team who will run it on Windows, where such processing is unwieldy at best.

**Comment by hackery** [comment]
Thanks. Is the roadmap published anywhere?

**Comment by floragunncom** [comment]
Not yet, but will have soon something like this ...

**Comment by pavolloffay** [comment]
+1 on this issue

**Comment by floragunncom** [comment]
Support for environment variables in sg_config will come with the next release (v25)

**Comment by floragunncom** [comment]
Fixed with v25.0 which was released today 

```
${env.MY_ENV_VAR:-default}
${envbc.MY_ENV_VAR:-default} #this will use the env var value and bcrypt it
${envbase64.MY_ENV_VAR:-default} #this will use the env var value and base64 decode it 

or

${env.MY_ENV_VAR}
${envbc.MY_ENV_VAR}
${envbase64.MY_ENV_VAR}


```
